content/docs/services/tink.md

```diff
@@ -0,0 +1,205 @@
+
+### Build your own
+
+Luckily we use the standard Golang toolchain. You can clone the tink repository:
+
+```
+$ git clone git@github.com:tinkerbell/tink
+```
+
+All the binaries are inside `cmd/tink-*` based on what you need you can run `go
+build`. For example if you would like to compile the CLI:
+
+```
+$ go build cmd/tink-cli/main.go
+```
+
+You can even use `go run` if you want to just run code without having to compile
+a binary:
+
+```
+$ go run cmd/tink-server/main.go
+```
+
+### Docker images
+
+We relay on Docker a lot for code distribution but also for workflow execution.
+Our CI/CD pipeline build and push images to
+[quay.io](https://quay.io/tinkerbell) a popular image repository similar to
+Docker Hub.
+
+As you can see we have a repository for every tool:
+
+* [tink-cli](https://quay.io/repository/tinkerbell/tink-cli?tab=tags)
+* [tink-worker](https://quay.io/repository/tinkerbell/tink-worker?tab=tags)
+* [tink-server](https://quay.io/repository/tinkerbell/tink?tab=tags)
+
+The tag are composed as: `sha-<gitsha>`, where `gitsha` is the first 7 chars of
+a git commit. Only master commits are pushed to quay.io.
+
+## Vagrant setup and local cli
+
+Prerequisite: a bit of familiarity with `go build`, Go has to be installed.
+
+I find myself way more comfortable where I can run a CLI locally. During this
+exercise you will follow the [Vagrant Setup](/docs/setup/local-with-vagrant)
+tutorial, when you get to the section "Creating the Worker’s Hardware Data",
+don't do it and we will build and run the tink-cli locally.
+
+Let's follow the [Vagrant Setup](/docs/setup/local-with-vagrant).
+
+Now that you have a provisioner up and running you can clone the tink
+repository:
+
+```
+$ git clone git@github.com:tinkerbell/tink.git
+$ cd tink
+```
+
+You can run the cli directly via:
+
+```
+$ go run cmd/tink-cli/main.go
+tinkerbell CLI
+
+Usage:
+  tink [command]
+
+Available Commands:
+  hardware    tink hardware client
+  help        Help about any command
+  template    tink template client
+  workflow    tink workflow client
+
+Flags:
+  -f, --facility string   used to build grpc and http urls
+  -h, --help              help for tink
+  -v, --version           version for tink
+
+Use "tink [command] --help" for more information about a command.
+```
+
+Now let's compile it with:
+
+```
+$ go build -o tink cmd/tink-cli/main.go
+$ tink hardware list
+Error: undefined TINKERBELL_CERT_URL
+Usage:
+  tink hardware list [flags]
+
+Flags:
+  -h, --help   help for list
+
+Global Flags:
+  -f, --facility string   used to build grpc and http urls
+
+undefined TINKERBELL_CERT_UR
+```
+
+`undefined TINKERBELL_CERT_UR` notifies you about an authentication issue you
+are facing with the `tink-server`. All the traffic between tinkerbell services is encrypted via TLS.
+There are two environment variable that we have to use in order to authenticate
+the CLI:
+
+- `TINKERBELL_CERT_URL=http://127.0.0.1:42114/cert`
```

Comment:
is there a general, non-Vagrant form of this? It does look like the ports would be the same in a non-Vagrant setup, but what about the IP Address?

Reply:
This works everywhere, it explains how you can authenticate the CLI against a Tink-Server

Reply:
Cool, thanks. I edited this section to be a bit more general, let me know if it's still correct.

Reply:
It works!